spec/inputs/stdin_spec.rb

```diff
@@ -31,8 +32,8 @@
   end
 
   context 'ECS behavior', :ecs_compatibility_support do
-
-    subject { LogStash::Inputs::Stdin.new }
+    require "logstash/codecs/json"
+    subject { LogStash::Inputs::Stdin.new("codec" => LogStash::Codecs::JSON.new) }
```

Comment:
review note why codec change needs: the root cause for test failure is, we ca calling directly `process()` method of the plugin. For the real scenarios, `StdinChannel::Reader.new` or standard `$stdin` escapes the new line. So, in the test cases, changed plugin default `Line` codec to `JSON` codec to avoid the new line (`\n`).

Reply:
Instead of changing the codecs, maybe we can add `strip` to the validation
```
expect( event.get('event') ).to eql 'original' => stdin_data.strip
```
just like the "sets hostname" test case